tests/dragonfly/utility.py

```diff
@@ -91,22 +92,31 @@ async def info_tick_timer(client: aioredis.Redis, section=None, **kwargs):
         yield x
 
 
-async def wait_available_async(client: aioredis.Redis, timeout=120):
+# wait for replica to reach stable sync state
+async def wait_available_async(
+    clients: Union[aioredis.Redis, Iterable[aioredis.Redis]], timeout=120
+):
+    if not isinstance(clients, aioredis.Redis):
+        # Syntactic sugar to seamlessly handle an array of clients.
+        return await asyncio.gather(*(wait_available_async(c) for c in clients))
+
     """Block until instance exits loading phase"""
-    its = 0
+    # First we make sure that ping passes
     start = time.time()
     while (time.time() - start) < timeout:
         try:
-            await client.ping()
-            return
+            await clients.ping()
+            break
         except aioredis.BusyLoadingError as e:
             assert "Dragonfly is loading the dataset in memory" in str(e)
+    timeout -= time.time() - start
+    if timeout <= 0:
+        raise RuntimeError("Timed out!")
 
-        # Print W to indicate test is waiting for replica
-        print("W", end="", flush=True)
-        await asyncio.sleep(0.01)
-        its += 1
-    raise RuntimeError("Client did not become available in time!")
+    # Secondly for replicas, we make sure they reached stable state replicaton
+    async for info, breaker in info_tick_timer(clients, "REPLICATION", timeout=timeout):
+        with breaker:
+            assert info["role"] == "master" or "slave_repl_offset" in info, info
```

Comment:
Сould there be a case that we have a `info["role"] == slave` (or replica) but we don't have a full_sync (no slave_repl_offset) stage yet?

Reply:
It won't really matter, since we got `slave_repl_offset` in `info` which means that we reached `stable sync`

Reply:
My question is why we use this for both master and replica? Either the role is master or either the replica has reached stable sync. I find this a little confusing IMO

Reply:
@BagritsevichStepan yes, it can and that was the original bug. fullsync happens after the initial handshake between master and replica and it can take several milliseconds. during that time you have role == 'slave'
@kostasrim  wait_available_async is used for other use-cases to wait until master finishes loading its snapshot for example. I will update the comment.